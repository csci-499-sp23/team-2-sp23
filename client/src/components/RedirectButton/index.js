import { Button } from "@mui/material";
import LaunchIcon from "@mui/icons-material/Launch";

export default function RedirectButton({ image, url,  }) {
  const classes = {
    imageContainer: {
      height: "20px",
      objectFit: "cover",
      borderRadius: "2px",
    },
    buttonContainer: {
      height: "30px",
      display: "flex",
      justifyContent: "center",
      backgroundColor: "white",
      gap: "10px"
    },
  };
  return (
    <Button
      variant="outlined"
      style={classes.buttonContainer}
      onClick={() => {
        window.open(url, "_blank");
      }}
    >
      <img src={image} style={classes.imageContainer} alt="logo" />
      <LaunchIcon style={{height: "18px", width: "18px"}} />
    </Button>
  );
}
