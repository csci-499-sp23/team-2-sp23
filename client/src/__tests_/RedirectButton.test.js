import React from "react";
import renderer from "react-test-renderer";
import RedirectButton from "../components/RedirectButton";

test("renders correctly", () => {
  const tree = renderer.create(<RedirectButton />).toJSON();

  expect(tree).toMatchSnapshot();
});
