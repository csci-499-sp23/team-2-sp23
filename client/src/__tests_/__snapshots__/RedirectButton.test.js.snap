// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  RedirectButton
</button>
`;
